lib/state/data/middleware.ts

```diff
@@ -28,10 +29,18 @@ export const middleware: S.Middleware = (store) => (
         systemTags.splice(markdownInNote, 1);
       }
 
+      // apply selected tag by default
+      const selectedTag = state.ui.openedTag;
+      let tags = action.note?.tags ?? [];
+      if (selectedTag) {
+        tags = withTag(tags, selectedTag);
+        // tags.push(t(selectedTag));
+      }
```

Comment:
might be clearer to combine this into non-mutating expressions

```ts
const { openedTag } = state.ui;
const givenTags = action.note?.tags ?? [];
const tags = openedTag ? withTag(givenTags, openedTag) : givenTags;
```

Reply:
I've updated based on this recommendation and now using `const`s